Docs landing page

estheragbaje opened the pull request:
This PR contains the development of the docs landing page

- Adding sections based on the design
- Implementing responsiveness
- Accessibility
- Styling
- Light and dark mode

phazonoverload committed:
Custom Homepage Layout

phazonoverload committed:
Added link to other page

estheragbaje committed:
docs: add icons

estheragbaje committed:
docs: add assets

estheragbaje committed:
docs: create usecase component

estheragbaje committed:
docs: create tabs component

estheragbaje committed:
docs: create self hosting component

estheragbaje committed:
docs: remove hero component

estheragbaje committed:
docs: create footer component

estheragbaje committed:
docs: create docs card component

estheragbaje committed:
docs: create code toggler component

estheragbaje committed:
docs: create article component

estheragbaje committed:
docs: update packages

estheragbaje committed:
docs: update home

estheragbaje committed:
docs: update styles

rijkvanzanten committed:
Merge branch 'main' into docs-landing-page

rijkvanzanten committed:
Replace doccards on homepage

rijkvanzanten committed:
Install typescript

rijkvanzanten committed:
Use docs Card instead of DocCard

rijkvanzanten committed:
Standardize Tabs component to be globally available

rijkvanzanten committed:
Register Tabs globally

rijkvanzanten committed:
Update lockfile

rijkvanzanten committed:
Fix card margins

rijkvanzanten committed:
Tweaks

rijkvanzanten committed:
Merge branch 'main' into docs-landing-page

rijkvanzanten committed:
Merge branch 'main' into docs-landing-page

estheragbaje committed:
docs: remove icon svgs

estheragbaje committed:
docs: update components

estheragbaje committed:
docs: add divider component

estheragbaje committed:
docs: update accessibility

estheragbaje committed:
docs: fetch featured articles

estheragbaje committed:
docs: fetch featured articles

estheragbaje committed:
docs: update home page

estheragbaje committed:
docs: update home page

estheragbaje committed:
docs: update package

estheragbaje committed:
docs: update package

estheragbaje committed:
docs: update padding

estheragbaje committed:
docs: update padding

rijkvanzanten commented on docs/guides/headless-cms/schedule-content/index.md:
```diff
@@ -1,13 +1,17 @@
-  url="/guides/headless-cms/schedule-content/dynamic-sites" />
+  text="Learn how to conditionally show content on dynamic websites."
+  url="/guides/headless-cms/schedule-content/dynamic-sites"
+  add-margin
```
The `addMargin` prop was removed in https://github.com/directus/directus/pull/18911/files#diff-7f7a2886ba028429230a8f3655f1c8b51e6b62ee8b08f14d14bbe7745a22f40eL16

rijkvanzanten commented on docs/public/assets/baas.png:
These should come from the Directus instance

estheragbaje committed:
Merge branch 'docs-landing-page' of https://github.com/directus/directus into docs-landing-page

estheragbaje committed:
docs: update secrets

estheragbaje committed:
docs: remove redundant articles component

estheragbaje committed:
docs: create tag component

phazonoverload committed:
Removed authenticated request

phazonoverload committed:
Merge commit

estheragbaje committed:
refactor: article component

estheragbaje commented on docs/.vitepress/components/Divider.vue:
```diff
@@ -0,0 +1,10 @@
+<template>
+	<div class="divider" />
```
Considering that the divider serves a purely visual or presentational role, I'm not too sure we should prioritize a11y regarding it.

estheragbaje commented on docs/.vitepress/components/home/UseCase.vue:
```diff
@@ -0,0 +1,48 @@
+<template>
+	<div class="use-case">
+		<a :href="url">
+			<img :src="img" alt="" class="img-size" />
```
Combined this

estheragbaje commented on docs/.vitepress/components/home/SelfHosting.vue:
```diff
@@ -0,0 +1,36 @@
+<template>
+	<div class="sh">
+		<img :src="img" alt="" class="sh-img" />
```
I'll suggest we keep this as a different component entirely because unlike the article, this has a button

estheragbaje commented on docs/.vitepress/components/home/Articles.vue:
```diff
@@ -0,0 +1,3 @@
+<template>
+	<p>{{ data }}</p>
```
This was a redundant component. Taken it out

estheragbaje committed:
refactor: api composition

estheragbaje committed:
feat: create social icon component

estheragbaje committed:
docs: update footer

estheragbaje committed:
chore: update icons

estheragbaje committed:
feat: update component styling

estheragbaje committed:
docs: update copy

estheragbaje committed:
chore: page cleanup

estheragbaje committed:
docs: update tab styling

estheragbaje committed:
docs: update hero styling

estheragbaje committed:
docs: update styling

estheragbaje committed:
docs: update links

estheragbaje committed:
chore: remove previous code toggler

estheragbaje committed:
feat: add token for hero, footer bg

estheragbaje committed:
docs: update background color

estheragbaje committed:
chore: update color

estheragbaje committed:
docs: add hero svg pattern

estheragbaje committed:
docs: update urls

estheragbaje committed:
Merge branch 'main' of https://github.com/directus/directus into docs-landing-page

estheragbaje committed:
docs: update packages

estheragbaje committed:
chore: update images

estheragbaje committed:
fix: linting errors

estheragbaje committed:
fix: pattern blocking button

estheragbaje committed:
Add Snippet Toggler to Docs (#18950)

* feat: add snippet toggler

* docs: update styling

* chore: remove previous code toggler

* docs: update styling

* docs: update syling

* docs: update syntax code theme highlighting

* docs: remove styles

* docs: update styles

* docs: update packages

* chore: remove initial

* feat: persist snippet toggler value

* docs: update theme

* chore: reduce jank for snippet toggler

estheragbaje committed:
fix: update spell checker

phazonoverload commented:
Just working through with a fine toothcomb - mostly related to links.

- Getting Started - I would send folks to the quickstart guide, which should be a focus in the next sprint. 
- Dev Reference -> Real Time link is broken. Also, we've been stylizing it 'Realtime' as a product name. 
- All User Guide links are broken but I think this is assuming the user guide PR is merged. 
- Graphic for config options doesn't make sense. Is this planned to be updated or shall we whip something up quickly? 

Final tiny design thing - any chance we can stop the nav between dev reference and user guide jumping when an option is clicked? 

This looks really, really good.

estheragbaje committed:
chore: update url

rijkvanzanten reviewed (dismissed):
![CleanShot 2023-06-26 at 14 56 44@2x](https://github.com/directus/directus/assets/9141017/d81f1c09-6333-47ca-af85-8540355a2545)

Things seem to fall apart a little bit on the in-between breakpoint

phazonoverload commented:
> Things seem to fall apart a little bit on the in-between breakpoint

I reckon we should nix the snippet at this size

rijkvanzanten commented:
> > Things seem to fall apart a little bit on the in-between breakpoint
> 
> I reckon we should nix the snippet at this size

Yeah. A quick(er) fix would be to kick in the mobile breakpoint sooner, but it's nicer to have a proper in-between set of styles 👍🏻

rijkvanzanten committed:
Remove homepage redirect

estheragbaje committed:
fix: update color variables

estheragbaje committed:
fix: responsiveness of snippet toggler

estheragbaje committed:
Merge branch 'docs-landing-page' of https://github.com/directus/directus into docs-landing-page

estheragbaje committed:
chore: update asset

estheragbaje committed:
Merge remote-tracking branch 'origin' into docs-landing-page

paescuj committed:
Merge branch 'main' into docs-landing-page

rijkvanzanten committed:
Move home styles to separate file

rijkvanzanten committed:
tweak hero hover

rijkvanzanten committed:
Hero snippet tweaks

rijkvanzanten committed:
Finalize header

rijkvanzanten committed:
Tweak tab spacing

rijkvanzanten committed:
Styling tweaks

rijkvanzanten committed:
Move global styles to global overrides

paescuj committed:
Remove leftover deps

paescuj committed:
TS issues and small clean-ups

paescuj committed:
Commit missing tsconfig

paescuj committed:
Finalize tsconfig

paescuj committed:
Enhance border-radius for snippet toggler & tab button

Also fix warning about font-feature-settings

paescuj committed:
Fix bug in Firefox with drop-shadow

rijkvanzanten committed:
Merge branch 'main' into docs-landing-page

paescuj committed:
Fix lint issues

paescuj committed:
prettier-ignore no longer necessary, thx to new lint/format setup

paescuj committed:
Merge branch 'main' into docs-landing-page

paescuj committed:
@directus/tsconfig not suitable for Vue

rijkvanzanten committed:
Start on footer

rijkvanzanten committed:
Fix responsiveness of footer

paescuj committed:
Merge branch 'main' into docs-landing-page

rijkvanzanten committed:
Couple more responsive tweaks

rijkvanzanten committed:
Tweaks for paescuj

paescuj committed:
Prevent height change of hero when toggeling snippets

rijkvanzanten committed:
Add hover state to article card

paescuj committed:
Make select arrow clickable & hover effect

paescuj committed:
Remove obsolete classes & add margin in self-hosted section

paescuj committed:
Align indents

paescuj committed:
Fix Article components

paescuj committed:
Enforce comp order in Vue files to find things easier

paescuj committed:
Replace docker image by official one

paescuj committed:
Use Article component for self-hosted section & other clean-ups

paescuj committed:
Workaround for navbar on smaller screens like iPad

rijkvanzanten committed:
Merge branch 'main' into docs-landing-page

paescuj committed:
Link to docker quickstart guide instead of full guide

Co-authored-by: Kevin Lewis <kvn@lws.io>

rijkvanzanten committed:
Merge branch 'main' into docs-landing-page

paescuj committed:
Various optimization

- Use module style, see https://vitepress.dev/guide/using-vue#script-and-style
- Clean-up unused code / packages, ... - too distracting and can be
  restored through git
- Fix navbar workaround (apply style only on home page)
- Own component for button
- SSG fix in SnippetToggler
- Remove tags from articles
- Make snippet toggler in hero stand out a tiny little bit more
- Reduce max width in hero sub

paescuj committed:
Difference between internal / external links

paescuj committed:
Don't center self hosted header on larger screeens
